src/savedata.c

```diff
@@ -105,268 +34,265 @@ static void sub_02025C1C(s32 param0, int param1);
 
-    switch (v1) {
-    case 1:
-        v0->unk_0C = 0;
-    case 2:
-        sub_02024DBC(v0);
+    saveData->loadCheckStatus = 0;
 
-        v0->unk_04 = 1;
-        v0->unk_08 = 0;
+    switch (loadResult) {
+    case LOAD_RESULT_OK:
+        saveData->fullSaveRequired = FALSE;
+    case LOAD_RESULT_CORRUPT:
+        sub_02024DBC(saveData);
 
-        if (v1 == 2) {
-            v0->unk_10 |= (1 << 0);
+        saveData->dataExists = TRUE;
+        saveData->isNewGameData = FALSE;
+
+        if (loadResult == LOAD_RESULT_CORRUPT) {
+            saveData->loadCheckStatus |= NORMAL_LOAD_CORRUPT;
         }
 
         {
-            int v2, v3;
+            int frontierResult, videoResult;
 
-            sub_02024CD4(v0, &v2, &v3);
+            SaveDataExtra_LoadCheck(saveData, &frontierResult, &videoResult);
 
-            if (v2 == 3) {
-                v0->unk_10 |= (1 << 3);
-            } else if (v2 == 2) {
-                v0->unk_10 |= (1 << 2);
+            if (frontierResult == LOAD_RESULT_ERROR) {
+                saveData->loadCheckStatus |= FRONTIER_LOAD_ERROR;
+            } else if (frontierResult == LOAD_RESULT_CORRUPT) {
+                saveData->loadCheckStatus |= FRONTIER_LOAD_CORRUPT;
             }
 
-            if (v3 == 3) {
-                v0->unk_10 |= (1 << 5);
-            } else if (v3 == 2) {
-                v0->unk_10 |= (1 << 4);
+            if (videoResult == LOAD_RESULT_ERROR) {
+                saveData->loadCheckStatus |= VIDEO_LOAD_ERROR;
+            } else if (videoResult == LOAD_RESULT_CORRUPT) {
+                saveData->loadCheckStatus |= VIDEO_LOAD_CORRUPT;
             }
         }
         break;
-    case 3:
-        v0->unk_10 |= (1 << 1);
-    case 0:
-        sub_0202479C(v0);
+    case LOAD_RESULT_ERROR:
+        saveData->loadCheckStatus |= NORMAL_LOAD_ERROR;
+    case LOAD_RESULT_EMPTY:
+        SaveData_Clear(saveData);
         break;
     }
 
-    return v0;
+    return saveData;
 }
 
-SaveData * sub_020245A4 (void)
+SaveData* SaveData_Ptr (void)
 {
-    GF_ASSERT(Unk_021C0794 != NULL);
-    return Unk_021C0794;
+    GF_ASSERT(sSaveDataPtr != NULL);
+    return sSaveDataPtr;
 }
 
-void * SaveData_Get (SaveData * param0, int param1)
+void* SaveData_SaveTable (SaveData *saveData, int saveTableID)
 {
-    GF_ASSERT(param1 < 38);
-    return &(param0->unk_14.unk_00[param0->unk_20024[param1].unk_08]);
+    GF_ASSERT(saveTableID < SAVE_TABLE_ENTRY_MAX);
+    return &(saveData->body.data[saveData->pageInfo[saveTableID].location]);
 }
 
-const void * sub_020245E0 (const SaveData * param0, int param1)
+const void* SaveData_SaveTableConst (const SaveData *saveData, int saveTableID)
 {
-    return SaveData_Get((SaveData *)param0, param1);
+    return SaveData_SaveTable((SaveData*)saveData, saveTableID);
 }
 
-BOOL sub_020245E8 (SaveData * param0)
+BOOL SaveData_Erase (SaveData *saveData)
 {
-    int v0, v1;
-    u8 * v2 = Heap_AllocFromHeapAtEnd(3, 0x1000);
+    int sectorIndex, unused;
```

Comment:
Is `unused` not stripped by the compiler? Does it not match if you remove the extraneous var?

Additionally, can you move `sectorIndex` further down?